fix : update dockerfile for  homebrew linux changes

oadedayo-ch opened the pull request:
update Dockerfile command for homebrew linux install and adjusted environment variables to match the supported installation approach

oadedayo-ch committed:
fix : update dockerfile for  homebrew linux changes

sienkin commented on Dockerfile:
```diff
@@ -16,10 +16,18 @@ RUN amazon-linux-extras enable ruby3.0 && \
     yum clean all
 
-RUN /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+RUN mkdir -p /home/linuxbrew/.linuxbrew && \
+    curl -L https://github.com/Homebrew/brew/tarball/HEAD \
+    | tar xz --strip 1 -C /home/linuxbrew/.linuxbrew
```
As this `RUN` contains processing through a pipe, you should set the `SHELL` option `-o pipefail` before the `RUN` to ensure correct error handling.

E.g.:
```
SHELL ["/bin/bash", "-o", "pipefail", "-c"]

RUN mkdir -p ....
```

Additionally, this `RUN` and the `RUN` after it should be combined.

sienkin replied:
This updated approach does not not lead to a functioning Brew installation.

E.g.:
```
bash-4.2# brew install tfenv
hint: Using 'master' as the name for the initial branch. This default branch name
hint: is subject to change. To configure the initial branch name to use in all
hint: of your new repositories, which will suppress this warning, call:
hint:
hint: 	git config --global init.defaultBranch <name>
hint:
hint: Names commonly chosen instead of 'master' are 'main', 'trunk' and
hint: 'development'. The just-created branch can be renamed via this command:
hint:
hint: 	git branch -m <name>
Initialized empty Git repository in /home/linuxbrew/.linuxbrew/.git/
fatal: not in a git directory
Error: Need to download https://formulae.brew.sh/api/formula.jws.json but cannot as root! Run `brew update` without `sudo` first then try again.

bash-4.2# brew update
hint: Using 'master' as the name for the initial branch. This default branch name
hint: is subject to change. To configure the initial branch name to use in all
hint: of your new repositories, which will suppress this warning, call:
hint:
hint: 	git config --global init.defaultBranch <name>
hint:
hint: Names commonly chosen instead of 'master' are 'main', 'trunk' and
hint: 'development'. The just-created branch can be renamed via this command:
hint:
hint: 	git branch -m <name>
Initialized empty Git repository in /home/linuxbrew/.linuxbrew/.git/
fatal: not in a git directory
```

Does the "final" container also need the `linuxbrew` user to be added and the container run as that user?

oadedayo-ch replied:
I've set the pipefail option at the SHELL level in both stages, which gives us better error handling across all the piped commands. 
I've also added the user creation in the final stage and switched the container to run as that user

sienkin commented on Dockerfile:
```diff
@@ -44,8 +52,7 @@ RUN mkdir -p /home/linuxbrew > /dev/null 2>&1
 
-RUN /home/linuxbrew/.linuxbrew/bin/brew shellenv >> $BASH_ENV && \
-    /home/linuxbrew/.linuxbrew/bin/brew cleanup --prune=all
+RUN echo 'export PATH="/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin:$PATH"' >> "$BASH_ENV"
```
This `RUN` statement should be combined with the one after it.

oadedayo-ch committed:
fix: implement code review changes for homebrew Dockerfile:
- Added pipefail option for better error handling
- Container now runs as linuxbrew user
- Fixed file ownership and paths
